tests/cpp/test_allocation_domain.cpp

```diff
@@ -1423,4 +1423,19 @@ TEST_F(AllocationDomainTest, ReductionVectorization) {
   testValidate(executor_cache.fusion(), cg_outputs, inputs, __LINE__, __FILE__);
 }
 
+TEST_F(AllocationDomainTest, ClearReductionIterDomainsPatch) {
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+  auto tv0 = TensorViewBuilder()
+                 .ndims(3)
+                 .shape({-1, 1, -1})
+                 .contiguity({true, std::nullopt, true})
+                 .build();
+  auto tv1 = sum(tv0, {0});
+  tv1->setAllocationDomain(
+      {tv1->axis(0), tv1->axis(2), tv1->axis(1)},
+      {std::nullopt, true, std::nullopt});
+  tv1->clearReductionIterDomains();
```

Comment:
Is it possible to come up with a better test? I assume the user of nvFuser API (Python or csrc/ops) never calls clearReductionIterDomains. So the test as is seems to be testing an internal implementation that's subject to change, not an API behavior.

Reply:
> Is it possible to come up with a better test?

I like having a tight-fitting test to go with a patch PR.

`TensorView::clearReductionIterDomain` is a public API. So I think it's fair to have this as a *test that there's no assert* thing. But I agree that it's not that interesting a thing to put here. But as Naoya suggested below, if I put validation here, then it's a bad test on asserting internal implementation.

So I agree with @wujingyue here, I'll dig up the other PR for an end-2-end test that goes through `FusionExecutorCache`.

Reply:
I'm not sure if I understand here correctly, but why not just validating the root and allocation domains are set as intended?

Reply:
> I'm not sure if I understand here correctly, but why not just validating the root and allocation domains are set as intended?

I think for this specific API, it's a fairly reasonable validation. i.e. Reduction should be dropped and contiguity flags updated as well. @wujingyue was arguing that `clearReductionIterDomains` is not a publicly exposed API. So the test isn't meaningful. Does adding the validation suggested by @naoyam make more sense for us to keep the simple test?

Reply:
I'm OK with the test as is. My only complaint was/is that this test can be fragile *if* we later want to use a different way to process the output IterDomains for Welford. We might not even have a counterpart API to test against.

Reply:
> My only complaint was/is that this test can be fragile if we later want to use a different way to process the output IterDomains for Welford. We might not even have a counterpart API to test against.

Yeah that part is understood and I agree that we should have better test coverage for reduction/normalization scheduler in general.

The issue patched in this PR is just one of the issues discovered in the refactor work #2168.
I have #2202 to track patching allocation domain support in reduction scheduler, which I think would be a better place to put functional end-2-end tests with the scheduler.